textile/chat-features.textile

```diff
@@ -188,19 +188,19 @@ As well as user-initiated operations, the room must monitor its underlying resou
 ** @(CHA-RL4a)@ The state monitor must handle @UPDATE@ events from each contributors underlying Realtime Channel
 *** @(CHA-RL4a1)@ @[Testable]@ If the @resumed@ flag of the update is set to @true@, then no action should be taken (i.e. @CHA-RL4a3@ and @CHA-RL4a4@ behaviours are not performed).
 *** @(CHA-RL4a2)@ @[Testable]@ If the given contributor has not yet successfully managed to attach its Realtime Channel (i.e. no call to @attach()@ on the channel, per @CHA-RL1f@, has succeeded), then no action should be taken (i.e. @CHA-RL4a3@ and @CHA-RL4a4@ behaviours are not performed).
-*** @(CHA-RL4a3)@ @[Testable]@ If a room lifecycle operation is in progress, then a pending discontinuity event shall be recorded for this contributor - though it must not overwrite any existing discontinuity event. The @ErrorInfo@ associated with the discontinuity shall be the @reason@ for the underlying channel state change. The event will be notified to the contributor at a later point, as noted in this specification.
-*** @(CHA-RL4a4)@ @[Testable]@ If a room lifecycle operation is not in progress, then a discontinuity event will immediately be emitted to the contributor. The @ErrorInfo@ associated with the discontinuity shall be the @reason@ for the underlying channel state change.
+*** @(CHA-RL4a3)@ @[Testable]@ If a room lifecycle operation is in progress for the given room, then a pending discontinuity event shall be recorded for this contributor - though it must not overwrite any existing discontinuity event. The @ErrorInfo@ associated with the discontinuity shall be the @reason@ for the underlying channel state change. The event will be notified to the contributor at a later point, as noted in this specification.
```

Comment:
```suggestion
*** @(CHA-RL4a3)@ @[Testable]@ If a room lifecycle operation is in progress and has pending operations as per (queued as per CHA-RL7), then given discontinuity event shall be recorded for this contributor - though it must not overwrite any existing discontinuity event. The @ErrorInfo@ associated with the discontinuity shall be the @reason@ for the underlying channel state change. The event will be notified to the contributor at a later point, as noted in this specification.
```

Reply:
This is to conform to room atomicity as per CHA-RL7 and events emitted by ongoing operation, captured by room monitoring shouldn't affect pending queued operations WDYT

Reply:
I strongly all atomic `RETRY`, `RELEASE`, `ATTACH OR DETACH` operations takes precedence over `Room monitoring`, hence room monitoring events triggered as a side effect of any of those operations shouldn't cause any side effect in future pending atomic operations.

Reply:
This can be `true` for all `room monitoring` spec points

Reply:
In `chat-js`, 
```
private _operationInProgress = false;

```
can simply be updated to 
```
private operationInProgress = () -> !_mtx.isLocked()
```
**Changing code to above will result in the exact same behaviour.**

At the same time, we will have better readability and will significantly reduce current/future efforts of setting/resetting `_operationInProgress` flag.

Reply:
[Link to isLocked() impl](https://github.com/DirtyHairy/async-mutex/blob/8e38062a7f20ddf30143c4bce23ef8a672136ed6/src/Semaphore.ts#L62-L64), it basically return `true` if there are ongoing operation, otherwise `false` if no ongoing operations.